### PR TITLE
Fix crash on umd and module merge, allow umds to be accessed when merged with a non-UMD symbol

### LINKED ISSUE #28762: [BUG] Error: Debug Failure. Invalid cast. The supplied value [object Object] did not pass the test 'isJsonSourceFile'.
**TypeScript Version:**  3.2.1

**Search Terms:**

**Code**

```ts
declare global {
  const React: typeof import("react");
  const ReactDOM: typeof import("react-dom");
}

export {};
```

**Expected behavior:**

**Actual behavior:**
```
> tsc

/Users/akari/Developer/test/node_modules/typescript/lib/tsc.js:71026
                throw e;
                ^

Error: Debug Failure. Invalid cast. The supplied value [object Object] did not pass the test 'isJsonSourceFile'.
    at Object.cast (/Users/akari/Developer/test/node_modules/typescript/lib/tsc.js:1158:22)
    at getTypeOfVariableOrParameterOrPropertyWorker (/Users/akari/Developer/test/node_modules/typescript/lib/tsc.js:29354:41)
    at getTypeOfVariableOrParameterOrProperty (/Users/akari/Developer/test/node_modules/typescript/lib/tsc.js:29334:48)
    at getTypeOfSymbol (/Users/akari/Developer/test/node_modules/typescript/lib/tsc.js:29603:24)
    at resolveImportSymbolType (/Users/akari/Developer/test/node_modules/typescript/lib/tsc.js:33159:45)
    at getTypeFromImportTypeNode (/Users/akari/Developer/test/node_modules/typescript/lib/tsc.js:33141:25)
    at getTypeFromTypeNode (/Users/akari/Developer/test/node_modules/typescript/lib/tsc.js:33460:28)
    at checkImportType (/Users/akari/Developer/test/node_modules/typescript/lib/tsc.js:43996:13)
    at checkSourceElement (/Users/akari/Developer/test/node_modules/typescript/lib/tsc.js:46921:28)
```
**Playground Link:** 
https://gist.github.com/magic-akari/aa10dfb809268eb4f63d5749d113a93a

**Related Issues:**

### PR DESCRIPTION
**Pull request by weswigham**
Fixes #28762

### CODE REVIEW

**Comment by chrisbenti** [comment]
Works for me 😄

File: tests/baselines/reference/exportAsNamespace_augment.errors.txt [hunk 1]
```diff
@@ -38,13 +34,5 @@
     }
     
     a.x + a.y + a.z + a.conflict;
-    ~
```
[the remark this answers is not in the document]
Comment (sheetalkamat) [reply tests/baselines/reference/exportAsNamespace_augment.errors.txt]: But aren't the globals accessible from module? They are only accessible from non module file?

**Comment by weswigham** [comment]
@danielrosenwasser since this fixes a crash, do we want to port it to 3.2?

**Comment by chrisbenti** [comment]
This is an active 3.2 blocker for a few of our repos so I'd very much appreciate it 😄

**Comment by xueyu2010** [comment]
+1 this is a blocker for my project as well, much appreciated if this fix can be released soon.  
Is there any interim fix I can avoid this exception?

**Comment by weswigham** [comment]
Yesterday's nightly should've included the fix, so installing `typescript@next` should work for the interim.

**Comment by xueyu2010** [comment]
> Yesterday's nightly should've included the fix, so installing `typescript@next` should work for the interim.

Thanks, works for me 😀

**Pull request merged**